fleetspeak/src/osquery/proto/fleetspeak_osquery/osquery.proto

```diff
@@ -0,0 +1,60 @@
+syntax = "proto3";
+
+package fleetspeak.osquery;
+
+enum CompressionType {
+  UNCOMPRESSED = 0;
+  ZCOMPRESSION = 1;
+}
+
+// A LoggedResult message contains data provided to an osquery Logger, containing
+// e.g. query results from a snapshot query.
+message LoggedResult {
+  enum Type {
+    UNKNOWN = 0;
+    STRING = 1;
+    SNAPSHOT = 2;
+    HEALTH = 3;
+    INIT = 4;
+    STATUS = 5;
+  }
+  Type type = 1;
+
+  CompressionType compress = 2;
+
+  // JSON payload, encoded according to the compress attribute.
+  bytes data = 3;
+}
+
+// A Queries message contains queries to execute using osquery through the
+// osquery distributed inteface.
+message Queries {
+  // map from query name to SQL text
+  map<string, string> queries = 1;
+
+  // map from query name to SQL text, if set the corresponding query will only
+  // be executed if the discory query returns a result.
```

Comment:
maybe be explicit that the name of the query should match in 'queries' and 'discovery'?

Something like: "if set, the query of the same name in 'queries' will only be executed if the corresponding discovery query returns a result." ?

Reply:
Tried to make the comment more explicit.